lib/src/main/java/org/esbtools/lightbluenotificationhook/NotificationHookConfiguration.java

```diff
@@ -6,55 +6,100 @@
+ *
+ * arrayOrderingSignificant: If this is set to true, and if the only
+ * thing changed in an array is the ordering of its elements, an event
+ * is still generated. If false, array re-orderings are
+ * ignored. Default is false.
+ */
 public class NotificationHookConfiguration implements HookConfiguration {
     private final Projection watchProjection;
     private final Projection includeProjection;
+    private final Projection propertiesProjection;
+    private final boolean arrayOrderingSignificant;
 
-    public NotificationHookConfiguration(Projection watchProjection, Projection includeProjection) {
+    public NotificationHookConfiguration(Projection watchProjection,
+                                         Projection includeProjection,
+                                         Projection propertiesProjection,
+                                         boolean arrayOrderingSignificant) {
         this.watchProjection = watchProjection;
         this.includeProjection = includeProjection;
+        this.propertiesProjection = propertiesProjection;
+        this.arrayOrderingSignificant = arrayOrderingSignificant;
     }
 
     public static <T> NotificationHookConfiguration fromMetadata(MetadataParser<T> parser,
-            T parseMe) {
+                                                                 T parseMe) {
         Projection watchProjection = parser.getProjection(parseMe, "watchProjection");
         Projection includeProjection = parser.getProjection(parseMe, "includeProjection");
-
-        return new NotificationHookConfiguration(watchProjection, includeProjection);
+        Projection propertiesProjection = parser.getProjection(parseMe, "propertiesProjection");
+        Object b=parser.getValueProperty(parseMe, "arrayOrderingSignificant");        
+        
+        return new NotificationHookConfiguration(watchProjection,
+                                                 includeProjection,
+                                                 propertiesProjection,
+                                                 b instanceof Boolean? (Boolean)b:false );
     }
-
+    
     public Projection watchProjection() {
         return watchProjection;
     }
-
+    
     public Projection includeProjection() {
         return includeProjection;
     }
 
+    public Projection propertiesProjection() {
+        return propertiesProjection;
+    }
+
+    public boolean isArrayOrderingSignificant() {
+        return arrayOrderingSignificant;
+    }
+
     public <T> void toMetadata(MetadataParser<T> parser, T writeMe) {
         parser.putProjection(writeMe, "watchProjection", watchProjection);
         parser.putProjection(writeMe, "includeProjection", includeProjection);
+        if(propertiesProjection!=null)
+            parser.putProjection(writeMe, "propertiesProjection", propertiesProjection);
+        if(arrayOrderingSignificant)
+            parser.putValue(writeMe,"arrayOrderingSignificant",Boolean.TRUE);
```

Comment:
Why only write these conditionally? IMHO write it either way to be consistent... this is less surprising to me. Not a huge deal though.